dspace-api/src/main/java/org/dspace/content/dao/impl/RelationshipDAOImpl.java

```diff
@@ -201,18 +201,24 @@ public int countRows(Context context) throws SQLException {
     }
 
     @Override
-    public int countByItemAndRelationshipType(Context context, Item item, RelationshipType relationshipType)
-            throws SQLException {
+    public int countByItemAndRelationshipType(Context context, Item item, RelationshipType relationshipType,
+                                              boolean isLeft) throws SQLException {
```

Comment:
I don't specially like the approach of passing parameters that in my perspective describe the property of an object. In this particular case the object Relationship. The concept of Left and Right item are always subjective and relative to the person that defines it. I think a different approach that could be considered in the future is to probably considers an Item not for the location/position which is it, but if it's itself in a relation verification.

But I understand the reason for having this boolean here. And considering the required effort for a different approach, I'm ok with having it this way in this version.

Reply:
I agree with @paulo-graca 's comment here.  It seems like this `isLeft` is needed, but it's an awkward implementation...the left/right concept is turning out to be quite confusing (and it is subjective).  I'm not sure what to replace it with though, as that would take more detailed discussion that would likely need to be delayed for post 7.0

Reply:
The Left/Right contents is indeed not pre-defined, and just dependent on how the relation is configured. The concept of Left/Right was chosen to make it easier if you draw the relationship on paper.
It's not an easy concept to support any relationship in code, but feel free to create a ticket with suggestions to make this easier to understand.
For 7.0 it's indeed not realistic to adjust this